packages/core/src/provider/initConfigs.ts

```diff
@@ -94,35 +99,53 @@ const initEnvironmentConfigs = (
   };
 
   if (environments && Object.keys(environments).length) {
-    return Object.fromEntries(
-      Object.entries(environments).map(([name, config]) => {
-        const environmentConfig: MergedEnvironmentConfig = {
-          ...(mergeRsbuildConfig(
-            {
-              ...rsbuildSharedConfig,
-              dev: {
-                assetPrefix,
-                lazyCompilation,
-              },
-            } as unknown as MergedEnvironmentConfig,
-            config as unknown as MergedEnvironmentConfig,
-          ) as unknown as MergedEnvironmentConfig),
-        };
-
-        return [name, applyEnvironmentDefaultConfig(environmentConfig)];
-      }),
+    const resolvedEnvironments = Object.fromEntries(
+      Object.entries(environments)
+        .filter(([name]) => isEnvironmentEnabled(name))
+        .map(([name, config]) => {
+          const environmentConfig: MergedEnvironmentConfig = {
+            ...(mergeRsbuildConfig(
+              {
+                ...rsbuildSharedConfig,
+                dev: {
+                  assetPrefix,
+                  lazyCompilation,
+                },
+              } as unknown as MergedEnvironmentConfig,
+              config as unknown as MergedEnvironmentConfig,
+            ) as unknown as MergedEnvironmentConfig),
+          };
+
+          return [name, applyEnvironmentDefaultConfig(environmentConfig)];
+        }),
+    );
+
+    if (!Object.keys(resolvedEnvironments).length) {
+      logger.error(
+        `The current project is specified to run only in the ${color.yellow(specifiedEnvironments?.join(','))} environment, but the configuration of the corresponding environment was not found.`,
+      );
+      process.exit(1);
```

Comment:
We prefer to throw an error instead of calling `process.exit(1)`. It is not recommended to call `process.exit()`.